### DISCUSSION TITLE
AGU Sunday 1/2 day hybrid workshop (Dec 10, 13:00-16:30) - Ideas and Tips

### DISCUSSION THREAD

**Discussion by michele-ornl** [NASA-Openscapes/2023-planning-agu#3] [Ideas]
Which tutorials should we present?

**Comment by stefaniebutland** [comment]
@michele-ornl has a tip for folks giving workshops. Showing people they can combine data on the cloud with local data. You might have a shapefile (a zip file or other data you have downloaded) …you can drag and drop from your local file system to a Hub folder!  

Outlined in her ESA 2023 notebook (in [2i2c Hub](https://openscapes.2i2c.cloud/hub/user-redirect/lab/tree/ESA_2023_ANG/CloudAVIRIS-NG_L2_examine.ipynb)) Step 2.1 Bringing your own data to the Cloud 

screenshots of drag & drop
![Screenshot 2023-09-28 at 11 38 19 AM](https://github.com/NASA-Openscapes/2023-planning-agu/assets/11927811/530dedbd-b1b6-468e-8389-10afd7c5646f)

**Comment by jules32** [comment]
We'll talk this through more tomorrow in the Cohort Call; Here (bookmarked below) is the refined agenda we submitted that will be published via AGU for signups: 

https://docs.google.com/document/d/1dOilx2mVi-HK4gout0SpYczyXpsZymL0h4bKaIQ-2ew/edit?pli=1#bookmark=id.ben7ag3fuplu

**Comment by BriannaLind** [comment]
We can adapt [this script](https://github.com/nasa/VITALS/blob/dev/python/01_Finding_Concurrent_Data.ipynb) that demonstrates concurrent data access of ECOSTRESS and EMIT data.  Catalina suggested potentially [OPERA inland water surface extent ](https://podaac.jpl.nasa.gov/dataset/OPERA_L3_DSWX-HLS_PROVISIONAL_V0)

**Comment by ScienceCat18** [comment]
Suggested "agenda" for co-working 10/12 (we may not get to everything but hopefully at least the first 3-4 items):

- [ ] review EDC Clinic [tutorial ](https://nasa-openscapes.github.io/earthdata-cloud-cookbook/examples/Earthdata-cloud-clinic.html)(by Amy)
- [ ] identify a science use case that would bring multiple DAAC/discipline datasets together 
- [ ] identify cross-daac datasets (can be mix of on prem and cloud archived, and auxiliary files like boundary shapefiles per Michele's comment above)
- [ ] identify who will take lead to develop/provide respective daac pieces for the tutorial (multiple contributors)
- [ ] identify tutorial instructors and breakdown of sections to be taught - needed to AGU by mid-Nov
- [ ] identify non-tutorial instructors/leads - see additional ideas already captured [here](https://docs.google.com/document/d/1dOilx2mVi-HK4gout0SpYczyXpsZymL0h4bKaIQ-2ew/edit?pli=1#bookmark=id.l0qx4rud2ycx) 
       - [ ] Welcome & Open Science Mindset
       - [ ] Hub orientation
       - [ ] Q&A/Community discussion
       - [ ] Development of Survey &Next Opportunities section


Previous science use cases ideas from recent 2023 events:
- Historic winter storms over CA: IMERG precip, SMAP salinity, SMAP soil moisture (some [inspiration](https://podaac.jpl.nasa.gov/DataAction-2022-03-01-SMAP-Visualizing-Ocean-Salinity-as-Powerful-Storms-Wreak-Havoc-in-California) and recent (9/2023) Worldview [story](https://worldview.earthdata.nasa.gov/?v=-162.295938399713,0.9864076331360891,-84.86031845217389,67.96658522488934&l=Reference_Features_15m,Coastlines_15m,VIIRS_SNPP_CorrectedReflectance_TrueColor&lg=true&tr=atmospheric_rivers&t=2023-02-24-T10%3A00%3A00Z))
- Florida very warm ocean waters, algae blooms? (news [article](https://www.washingtonpost.com/weather/2023/07/25/florida-record-warm-ocean-climate/))
- MUR SST, air temps over FL?, Landsat algae 
- VT flooding: precip, water extent, soil moisture?
- Canada wildfires: veg disturbance, AQ, ?

**Reply by ScienceCat18** [reply]
Just went through the recent [Atmospheric Rivers Worldview Tour Story](https://worldview.earthdata.nasa.gov/?v=-162.295938399713,10.726792525294101,-84.86031845217389,58.226200332731324&l=Reference_Features_15m,Coastlines_15m,VIIRS_SNPP_CorrectedReflectance_TrueColor&lg=true&tr=atmospheric_rivers&t=2023-02-24-T10%3A00%3A00Z) and it would be a pretty powerful story to demo at the workshop: "bring (some of) the datasets _previewed_ in NASA Worldview into a jupyter notebook in the cloud to get you started on a _deeper exploration_ of the data" of this abnormal and impactful event. 

Datasets in the Worldview story we could bring into the demo tutorial workflow: 
- GES DISC IMERG precip rate
- NSIDC SMAP snow mass (this product is new to me)
- NSIDC SMAP root zone soil moisture
- PODAAC OPERA surface water extent (reservoir water level changes)
- LP HLS imagery of impacts to flora/fauna - superbloom

It covers CA and the US West Coast, and since AGU Fall Meeting is in SF this year, it also seems extra appropriate.

Thoughts?

**Reply by BriannaLind** [reply]
Cool!  Yeah, I like that idea a lot, and I think it has room to grow!

**Reply by cassienickles** [reply]
Agenda: 
(20 min) Welcome & Open Science Mindset - Cassie 
(20 min) Introduction to NASA Earthdata Cloud - Michele - cheatsheets, champions slide deck materials, EDS (easy way to see if something is cloud) https://nsidc.github.io/data_strategies_for_future_us/data_strategies_slides#/title-slide
(20 min) Tutorial: Orientation and setup in JupyterHub in the Cloud - Michele https://nasa-openscapes.github.io/earthdata-cloud-cookbook/workshops/setup.html - look at Aaron's slides from UWG
(30 min) Tutorial: In-cloud Science Workflow (part 1): search & discovery - Aaron
(15 min) Break	 
(60 min) Tutorial: In-cloud Science Workflow (part 2): access, subset & plot - Danny
(30 min) Q&A / Community Discussion - panel of people in the room that are helpers & speakers, can invite people from coiled in the room
(15 min) Closing (survey and next opportunities) - Michele - survey? google form? Amy has done some surveying. - ask Amy later

Deck for welcome & open science mindset - Bri has a deck, can revamp the Champions presentations too, Danny says TOPS slidedeck of ethos of open science (Bri hard passes) https://nsidc.github.io/data_strategies_for_future_us/data_strategies_slides#/title-slide

Testing: Rupesh & Michele
      Decide on datasets - NSIDC SMAP snow mass or MODIS snow cover, GES DISC IMERG precip rate -February, California! - may need to cut down on this area if it takes too long.
      testing on datasets - Rupesh can help!

Revamping EDC Clinic notebook with multi datasets - Rupesh & Michele & Danny & Aaron update notebook, Cassie & Julie can help with quarto

Adding in 2 line code of earthaccess with multi datasets when available - Luis

Quarto book for workshop - Julie starting architecture, Cassie will fill it out/update what needs changed

dry runs - Catalina, Stef

People: 
Danny - instruct/present
Julie - Helper
Erin - Helper
Cassie - instruct/present
Luis - participating in another workshop - Helper at most
Aaron F. - Helper
Michele - instruct/present
Catalina - before meeting prep
Bri - before meeting prep
Celia - before meeting prep
Nikki - tester before meeting prep
Rupesh - testing, before meeting prep

**Reply by cassienickles** [reply]
intro to jupyter hub Here's that link https://wiki.earthdata.nasa.gov/download/attachments/281319592/Hand%20on%20activity%20-%20Getting%20Started.docx?version=1&modificationDate=1696349034781&api=v2

**Reply by jules32** [reply]
## Workshop quarto book architecture

Repo: https://github.com/NASA-Openscapes/2023-Cloud-Workshop-AGU 
Book: https://nasa-openscapes.github.io/2023-Cloud-Workshop-AGU 

Note: the GitHub Action seems to work (green check after some tinkering) but the page hasn't redeployed in 3 hours so the book isn't showing up at this time. 

PS: Cassie you are amazing, thank you for all this work planning/coordinating, and Catalina for building that thorough agenda!

**Reply by rupesh2** [reply]
We have done few of data testing: https://github.com/rupesh2/atmospheric_rivers

**Reply by amfriesz** [reply]
Hey all, I'm reviewing the EDC Clinic notebook and going over the notebooks that @rupesh2 shared. The notebooks that @rupesh2 put together are nice and concise and show accessing data from the URLS vs using Harmony. The EDC Clinic notebook is about half Harmony examples. I'd like to remove the Harmony portions of the notebook and expand a bit on `earthaccess`. Does anyone have any objections to that?

**Reply by amfriesz** [reply]
Does anyone already have a vector file that we could use to demonstrate using an input vector file to search with for the atmospheric river use case? Or do we want to stick with just using a bounding box over California?

**Reply by cassienickles** [reply]
We don't have a vector file already, feel free to pick whatever you'd like and let the group know! AGU is in SF this year so if you need to make the bbox smaller at all, perhaps focus in on the bay area. Up to you.

**Reply by cassienickles** [reply]
Tutorials discussion
Aaron would like a geojson file of boundaries for region - Bri is making this, a slice of CA
**datasets:**
IMERG precipitation
SMAP soil moisture
VIIRS snow 

Story: Atmospheric rivers, so cool! Let's look at some data that may have been affected! Could compare between years to see the difference.

develop tutorial here: https://github.com/NASA-Openscapes/2023-Cloud-Workshop-AGU 

xarray - great way to reduce data! can reduce to an average or total quickly if the data is right. 

MODIS - probably should not use since not in cloud.

How do we get github usernames from participants? - reach out 3 weeks before for email list? - Ask AGU. Participants need an EDL and github username. 

plotting packages: geoviews, holoviews, matplotlib, hvplot, leaflet, folium

@ michele in slack with slide - Cassie will move to new location.

**Reply by cassienickles** [reply]
slides for the workshop: https://docs.google.com/presentation/d/13huOTE6EQUY84qzqiRbRdK8coB1ZooU2dDLv1OaNiqI/edit#slide=id.p

**Comment by michele-ornl** [comment]
Catalina Taglialatela

Dry Run #1 (1h to cover:)
Welcome & Open Science Mindset - Cassie
Introduction to NASA Earthdata Cloud - Michele
Tutorial: Orientation and setup in JupyterHub in the Cloud - Michele
Proposed date: week of Nov 6

Dry Run #2 (1h to cover:)
Tutorial: In-cloud Science Workflow (part 1): search & access - Aaron
Proposed date: week of Nov 13

Dry Run #3 (1h to cover:)
 Tutorial: In-cloud Science Workflow (part 2): subset & plot - Danny
Proposed date: week of Nov 27

Final checkin (1-1.5 h if needed)
all supporting the workshop
Proposed date: week of Dec 4

An idea is to flip dry run dates and put the tutorials first if you prefer to have feedback earlier on in the design process.
I can send a poll for respective dry runs to make sure we settle on a date/time that works for instructors for the proposed weeks, if they look generally ok.

**Reply by cassienickles** [reply]
Nov 9th coworking - dry run #1 - michele & Cassie
Nov 15th mentors call - dry run #2 - Aaron & Danny both (doesn't have to be polished)
Nov 29th mentors call - dry run #3 - if still needed.
1st week of december, poll for M-W! See slack. @cassienickles send Erin times

Note: will gov't shutdown potential affect us in any way? - if shutdown happens, Nov 17th.

**Comment by BriannaLind** [comment]
geojson file: San Francisco ...... I couldn't attach so pasted below

{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {},
      "geometry": {
        "coordinates": [
          [
            [
              -122.5070795995562,
              37.17343508170461
            ],
            [
              -118.13461748565543,
              37.1521447785003
            ],
            [
              -118.17023835827376,
              38.94722002943203
            ],
            [
              -123.86067275905293,
              38.98184064006645
            ],
            [
              -122.5070795995562,
              37.17343508170461
            ]
          ]
        ],
        "type": "Polygon"
      }
    }
  ]
}

**Reply by amfriesz** [reply]
Thanks. I'll push that out to the repo

**Comment by cassienickles** [comment]
Know Before you go email!! @BriannaLind, please send this out (or have AGU send) to workshop participants when registration closes on 12/1

Hello Workshop Participants!

Thank you for registering to attend the **Enabling Analysis in the Cloud Using NASA Earth Science Data** Workshop hosted by [NASA Openscapes](https://nasa-openscapes.github.io/) on Sunday, December 10, 2023 from 1pm-4:30pm PST [(UTC-7)](https://www.timeanddate.com/time/zones/pdt)! We have a workshop website set up [here](https://nasa-openscapes.github.io/2023-Cloud-Workshop-AGU/), feel free to explore!

**_Before the Workshop,_** please ensure you have done the following:

**1. Signed up for an Earthdata Login Account**

> An Earthdata Login account is required to access data, as well as discover restricted data, from the NASA Earthdata system. Thus, to access NASA data, you need Earthdata Login. Please visit https://urs.earthdata.nasa.gov/ to register and manage your Earthdata Login account. This account is free to create and only takes a moment to set up. Please note your username and password for use during the day of the workshop!

**2. Signed up for a GitHub Account if you do not already have one**

> A GitHub account is required to gain access to the provided 2i2c cloud computing platform. Please visit https://github.com/join to register and create a free GitHub account. 

**3. Send your GitHub username to us via this [Google Form](https://docs.google.com/forms/d/e/1FAIpQLSfir0YIPhY5AJ6lpO1qOo-Wm1gjl7nXPasPjTKD3OCIhQo82A/viewform?usp=sf_link)**

> Make sure you have sent your username to the workshop leaders via the Google Form for access to our cloud computing environment.

**4. Bring a Laptop or tablet to the Workshop**

> Participation in the exercises requires a laptop or tablet. Yes, a tablet works too! All participants will have access to a 2i2c Jupyter Lab instance running in AWS us-west 2.

Looking forward to seeing you all at the workshop!

Best,
The NASA Openscapes Team

**Reply by BriannaLind** [reply]
@cassienickles  GREAT!  Thanks for putting this together!! I will send this out on or near Dec. 2 :)

**Reply by jules32** [reply]
AWESOME!

**Reply by BriannaLind** [reply]
actually was about to add to my calendar and saw that Dec. 1 is a Friday.  I'll get the list from AGU on Friday (or Monday) and send the email Monday (Dec.4) or Tuesday depending on when I get a response :)

**Reply by cassienickles** [reply]
email has been sent out to participants!

**Comment by BriannaLind** [comment]
FYI - 2i2c support is being tracked here: https://github.com/2i2c-org/infrastructure/issues/3430

**Comment by cassienickles** [comment]
### Logistics/Questions for today's final check in meeting:

**Logistics**

- email to participants (Bri sent the list to us) - Bri, are you sending the email drafted above or should Cassie? - **Cassie will!**
- 7 virtual attendees, 20 in person!
- live demo versions! - Aaron & Danny will have their own copy up for live demo version - Aaron recommends spell checks
- Computer: whose? Aaron's! & Danny's! 
- get there by 12pm! Grab your lunch and meet us there :)
- online: share zoom once we have it! 
- agufallmeeting-23 slack channel!!

**Workshop website** 

- Are the tutorial PR's finalized/will they be by the workshop - will be done!! :)
- Catalina will run through notebooks! great idea for helpers! - later in the week better for Danny's, Aaron's is squared away.
- Cassie needs to embed the slides into quarto still

**Slides/schedule**

- There are a couple slides that say delete, are we ready to delete? want to make slides public.
- Michele will add a bit more slides for hub start up
- Julie - screenshot for email accepting team invite
- "where do I go from here?" slide - what offboarding materials do we point to? Are there any that exist yet?
- Survey isn't done yet - google form - https://docs.google.com/forms/d/1Ybg4L3ULqdrixumz-DoK5MDk_xPpzq4ZmPNbrbykS8M/edit - use this AGU one!! YES THANK YOU JULIE: copy of this right here: https://docs.google.com/forms/d/18eZPFzewVl0qOsnJytYhk8Wpj_n3nVG4QA8BVv9w6eA/edit
- Pull out the questions for Q & A, but send out survey

How's everyone feeling? :) I'M EXCITED!!

**Reply by ScienceCat18** [reply]
Minor suggestions for the slides: if @cassienickles or @michele-ornl have the chance, include the Earthdata Cloud logo (small) as a footer or header on each slide. I have 4 versions i can send if needed (don't have them).

**Comment by BriannaLind** [comment]
[celebrate]     Lind, Brianna (Contractor) reacted to your message:
________________________________
From: Cassie Nickles ***@***.***>
Sent: Tuesday, December 5, 2023 5:18:57 PM
To: NASA-Openscapes/2023-planning-agu ***@***.***>
Cc: Lind, Brianna (Contractor) ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NASA-Openscapes/2023-planning-agu] AGU Sunday 1/2 day hybrid workshop (Dec 10, 13:00-16:30) - Ideas and Tips (Discussion #3)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



email has been sent out to participants!

**Reply by BriannaLind** [reply]
ALSO ----- looks like registration starts just before the workshop starts -----> in that case it is alright to go straight to the room and register after the workshop is complete :) You can pick up your badge at Moscone West or South during the hours below

**West Level 1 Full Registration and Scan & Go**

·         Sunday, 10 December – 12pm to 6pm

·         Monday, 11 December – 6:30am to 6:30pm

·         Tuesday, 12 December – 7am to 6pm

·         Wednesday, 13 December – 7:30am to 6pm

·         Thursday, 14 December – 7:30am to 6pm

·         Friday, 15 December – 7:30am to 5pm

**South Lobby Scan & Go** 

·         Sunday, 10 December – 12pm to 6pm

·         Monday, 11 December – 6:30am to 5pm

**Reply by jules32** [reply]
Good question. Since there are 8am workshops too, I am assuming we can teach before registering! I plan to add time for figuring this out before arrival just in case.
